packages/oauth/src/state-stores/file-state-store.ts

```diff
@@ -0,0 +1,108 @@
+import { homedir } from 'os';
+import fs from 'fs';
+import path from 'path';
+import { randomUUID } from 'crypto';
+import { StateStore, StateObj } from './interface';
+import { InstallURLOptions } from '../install-url-options';
+import { InvalidStateError } from '../errors';
+
+export interface FileStateStoreArgs {
+  stateExpirationSeconds?: number;
+  baseDir?: string;
+}
+
+export class FileStateStore implements StateStore {
+  private baseDir: string;
+
+  private stateExpirationSeconds: number;
+
+  public constructor(args: FileStateStoreArgs) {
+    this.baseDir = args.baseDir !== undefined ?
+      args.baseDir :
+      `${homedir()}/.bolt-js-oauth-states`;
+    this.stateExpirationSeconds = args.stateExpirationSeconds !== undefined ?
+      args.stateExpirationSeconds :
+      600;
+  }
+
+  public async generateStateParam(
+    installOptions: InstallURLOptions,
+    now: Date,
+  ): Promise<string> {
+    const state = randomUUID();
+    const source: StateObj = {
+      installOptions,
+      now,
+    };
+    this.writeToFile(state, source);
+    return state;
+  }
+
+  public async verifyStateParam(
+    now: Date,
+    state: string,
+  ): Promise<InstallURLOptions> {
+    try {
+      if (this.findFile(state)) {
+        // decode the state using the secret
+        let decoded: StateObj | undefined;
+        try {
+          decoded = this.readFile(state);
+        } catch (e) {
+          const message = `Failed to load the data represented by the state parameter (error: ${e})`;
```

Comment:
Perhaps I am mistaken but I don't think this `catch` will ever run?

Reply:
Ah, true. I added `this.findFile(state)` later on. I can remove the `this.findFile(state)` from this code.